### PR TITLE
Add mapping in r tips

### PR DESCRIPTION
**Pull request by cjrace**
## Overview of changes

Adding some notes on making maps in R that I started when editing the map app.

## Why are these changes being made?

Might be helpful for analysts to shortcut the initial learning process and have some handy immediate references to start from.

## Detailed description of changes

Added a variety of notes on mapping in R, an example interactive map in Shiny and tips for improving the load times of interactive maps.

There's definitely more information that could be added, though for now I figured I'd add the notes I had off the top of my head and we can always expand on demand.

I also renamed the project file as it's been bugging me I forgot to do it when we first moved over from the statisticians guide repo.

## Issue ticket number/s and link

No related issues.

## Checklist before requesting a review
- [x] I have checked the contributing guidelines
- [x] I have checked for and linked any relevant issues that this may resolve
- [x] I have checked that these changes build locally (checking this now)
- [x] I understand that if merged into main, these changes will be publicly available

### CODE REVIEW

**Comment by cjrace** [comment]
[Tidy up and PR notes for making maps in R](https://trello.com/c/yYokueuC/1396-tidy-up-and-pr-notes-for-making-maps-in-r)

File: learning-development/r.qmd [hunk 1]
```diff
@@ -389,6 +389,170 @@ To round numbers upwards in a way that is consistent with SQL and other programm
...
+
+---
+
+#### Boundary files
+
+---
+
+Most maps we'll want to make will be based on boundaries provided from the [Open Geography Portal](https://geoportal.statistics.gov.uk/).
+
+Take heed of the homepage and details on the codes for map resolution, usually the ultra generalised BUC boundaries are the best option for us as they still provide plenty of detail but are far smaller in size than the other formats.
```
Comment (jen-machin) [inline learning-development/r.qmd:402]: What is BUC?

File: learning-development/r.qmd [hunk 2]
```diff
@@ -389,6 +389,170 @@ To round numbers upwards in a way that is consistent with SQL and other programm
...
+
+If you want to edit the boundaries, whether that is to edit the coordinate system, file type or to simplify them at all, have a look at the [interactive website for mapshaper](https://mapshaper.org/), there's also an accompanying [rmapshaper R package](https://andyteucher.ca/rmapshaper/).
+
+---
+
+#### Packages
+
+---
+
+We always advise searching around the internet for any packages that might now exist when deciding to choose a package, though some known options for building maps are:
```
Comment (jen-machin) [inline learning-development/r.qmd:421]: This sentence is a bit clunky

File: learning-development/r.qmd [hunk 3]
```diff
@@ -389,6 +389,170 @@ To round numbers upwards in a way that is consistent with SQL and other programm
...
+        # Set the basemap (this is a good neutral one)
+        addProviderTiles(providers$CartoDB.PositronNoLabels) %>%
+        # Add the shaded regions
+        addPolygons(
+          color = "black",
+          weight = 1,
+          fillColor = pal_fun(map_data[["example_count"]]),
+          popup = map_popup
+        ) %>%
+        # Add a legend to the map
```
Comment (jen-machin) [inline learning-development/r.qmd:516]: I'm guessing this does not include the fix to the legend we need for accessibility 😄

File: learning-development/r.qmd [hunk 4]
```diff
@@ -389,6 +389,170 @@ To round numbers upwards in a way that is consistent with SQL and other programm
...
+
+2. Simplify the map detail
+
+If you're using a particularly detailed map, you can sometimes simplify the files used to help reduce the size, this will make the map less precise but have a trade off of speeding up the load times. See the [boundary files](#boundary-files) section for more information.
+
+3. Only change the layers you need to
+
+If the map layers on top are reactive (e.g. shaded polygons that change based on a user interaction like a dropdown), consider using the [leafletProxy function from the Leaflet package](https://rstudio.github.io/leaflet/articles/shiny.html#modifying-existing-maps-with-leafletproxy) to change only the layer you need to while leaving the base map static.
+
+4. If all else fails, try the magical hammer of caching
```
Comment (jen-machin) [inline learning-development/r.qmd:550]: 🤣

**Review by jen-machin — COMMENTED** [review]
Probably nitpicky comments and one entirely pointless one

File: learning-development/r.qmd (hunk 3 above)
Comment (cjrace) [reply learning-development/r.qmd:516]: You are correct, it does not 😅

File: learning-development/r.qmd (hunk 2 above)
Comment (cjrace) [reply learning-development/r.qmd]: Attempted to de-clunk now

**Review by jen-machin — APPROVED** [review]

**Pull request merged**